salt/modules/timezone.py

```diff
@@ -410,6 +410,12 @@ def set_hwclock(clock):
         elif clock == 'localtime':
             __salt__['file.sed']('/etc/default/rcS', '^UTC=.*', 'UTC=no')
     elif 'Gentoo' in __grains__['os_family']:
+        if clock not in ('UTC', 'localtime'):
+            raise SaltInvocationError(
+                'Only \'UTC\' and \'localtime\' are allowed'
+            )
+        if clock.__eq__('localtime'):
```

Comment:
@terminalmage Thank you for the nice comment. I appreciate it even more for the numbers.

I chose __eq__() method considering clock is a string object. I'm sorry I overlooked the speed issue and I'll follow up with a patch.